src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -279,6 +286,37 @@ public function addClassResource(\ReflectionClass $class)
     }
 
     /**
+     * Checks whether the requested class, interface or trait exists and registers the result for resource tracking.
+     *
+     * @param string $class
+     *
+     * @return bool
+     *
+     * @final
+     */
+    public function classExists($class)
+    {
+        if (isset($this->classReflectors[$class])) {
+            return (bool) $this->classReflectors[$class];
+        }
+        $exists = class_exists($class) || interface_exists($class, false) || trait_exists($class, false);
+
+        if ($this->trackResources) {
+            if (!$exists) {
+                $this->addResource(new ClassExistenceResource($class));
+                $this->classReflectors[$class] = false;
+            } else {
+                $class = $this->classReflectors[$class] = new \ReflectionClass($class);
```

Comment:
We don't need to store the reflection class, we can store `true` instead, right ?